### PR TITLE
Change rule expression to match endpoint types for receiver creator  

### PR DESCRIPTION
**Pull request by bjsignalfx**
**Description:** Change the rule expression in receiver creator for matching endpoints types **port**, **hostport** and **pod** from **type.port**, **type.hostport** and **type.pod** respectively, to **type == "port"**, **type == "hostport"** and **type == "pod"**.

**Testing:** Fixed unit tests failing due to change 

**Documentation:** Updated documentation to reflect change.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@jrcamp please review.

File: receiver/receivercreator/rules.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type rule struct {
 }
 
 // ruleRe is used to verify the rule starts type check.
-var ruleRe = regexp.MustCompile(`^type\.(pod|port)`)
+var ruleRe = regexp.MustCompile(`^type(\s*==\s*"pod"|\.port)`)
```
Comment (jrcamp) [inline receiver/receivercreator/rules.go:33]: While you're there can you also add "hostport", was missing.

File: extension/observer/endpoints.go [hunk 2]
```diff
@@ -74,6 +74,11 @@ func (e *Endpoint) Env() (EndpointEnv, error) {
 	env := e.Details.Env()
 	env["endpoint"] = e.Target
 
+	if e.Details.Type() == PodType {
```
Comment (jrcamp) [inline extension/observer/endpoints.go:77]: Should go ahead and change all types in one PR.

File: receiver/receivercreator/README.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ This setting controls what resource attributes are set on metrics emitted from t
 
 Note that the backticks below are not typos--they indicate the value is set dynamically.
 
-`type.pod`
+`type`
```
Comment (jrcamp) [inline receiver/receivercreator/README.md:65]: This doesn't provide the necessary context anymore. `type: "pod"` maybe?

File: receiver/receivercreator/README.md [hunk 4]
```diff
@@ -94,7 +94,7 @@ targeting it will have different variables available.
 
 | Variable    | Description                       |
 |-------------|-----------------------------------|
-| type.pod    | `true`                            |
+| type        | `pod`                            |
```
Comment (jrcamp) [inline receiver/receivercreator/README.md:97]:
```suggestion
| type        | `"pod"`                            |
```

**Review by jrcamp — CHANGES_REQUESTED** [review]

**Comment by tigrannajaryan** [comment]
Please resolve merge conflict.

**Review by jrcamp — APPROVED** [review]

**Pull request merged**